Composer design improvements

jancborchardt opened the pull request:
Lots of little fixes, but still some issues which be something for follow-up pull requests instead?

- [ ] Subject field is not automatically focused despite autofocus
- [ ] Possibly place "Draft saved" message below button? Also it could vanish after a few seconds?
- [ ] Sticky message options menu should open to the top → https://github.com/nextcloud/nextcloud-vue/issues/685
- [ ] Attachment upload buttons could go into the message options menu
- [ ] Attachment list layout needs to be rethought, especially on mobile → separate pull request I’d say
- [ ] Actions bar on bottom overlaps text, but if the previous 2 things are fixed then this is likely less of an issue.

## Before
![Mail composer before](https://user-images.githubusercontent.com/925062/67395028-89187080-f5a5-11e9-9601-6e47ee4a0319.png)

## After
![Mail composer after](https://user-images.githubusercontent.com/925062/67395029-89b10700-f5a5-11e9-9261-ea4b68b52582.png)


Please review @ChristophWurst @nextcloud/mail

ChristophWurst commented:
> [Work in progress]

You can also open a PR as draft now ;)

ChristophWurst commented on src/components/Composer.vue:
```diff
@@ -26,17 +29,19 @@
-			<a v-if="!showCC" href="#" @click.prevent="showCC = true">
-				{{ t('mail', '+ CC/BCC') }}
+			<a v-if="!showCC" class="transparency" href="#" @click.prevent="showCC = true">
+				{{ t('mail', '+ Cc/Bcc') }}
```
It was uppercase intentionally https://github.com/nextcloud/mail/pull/2071

jancborchardt replied:
So I changed it because both Gmail and Microsoft Outlook use "Cc" and "Bcc", and full-on uppercase just looks shouty.

And actually Apple Mail uses "Copy" and "Blind copy", which I would change it to instead as it is a lot less cryptic. :)

ChristophWurst replied:
cc @rakekniven

rakekniven replied:
@jancborchardt I checked older KMail and Thunderbird and Wikipedia.
Please change to other resources as well.

jancborchardt replied:
@rakekniven Wikipedia also uses "Cc" and "Bcc" (as well as of course explaining it stands for "Carbon copy" and "Blind carbon copy") https://en.wikipedia.org/wiki/Blind_carbon_copy

By other occurences do you mean in the Mail app, or is this elsewhere? Thanks for your help! :)

jancborchardt commented:
Updated and added before/after screenshot. There’s some possible todos, but in the sense of improving step by step we could do them in follow-up pull requests.

skjnldsv commented:
Maybe add a background on the bottom toolbar?

jancborchardt commented:
Added a fade from main background (bottom) to transparent (top) just like we do in the filepicker. As said above in the list though I’d rather fix the real issues there though.

jancborchardt commented:
Anything else? What is a blocker, what should I still fix, and what can be done separately @ChristophWurst?

ChristophWurst commented:
> Lots of little fixes, but still some issues which be something for follow-up pull requests instead?

Please open separate tickets for these things. Small PRs are good PRs :)

jancborchardt commented:
@ChristophWurst it's modeled after how the subject looks when reading mails too. I'd move from bold to regular, but the other changes make it lose structure since everything looks thr same.

(The alignment is intentional – it is aligned with the mail body.)

ChristophWurst commented:
> @ChristophWurst it's modeled after how the subject looks when reading mails too (as well as the general h2). I'd move from bold to regular, but the other changes make it lose structure since everything looks thr same.

isn't that what *gives* structure? If one of four things looks so different it breaks the structure IMO.

> 
> (The alignment is intentional – it is aligned with the mail body.)

Fair enough.

One more thing though, aren't we also introducing this anti-pattern of having only an inline label: https://baymard.com/blog/mobile-forms-avoid-inline-labels?

jancborchardt commented:
> isn't that what gives structure? If one of four things looks so different it breaks the structure IMO.

Sorry, structure was not really a good word to bring it across. I mean that it loses focus and guidance.
You would have 3 inputs below each other which look exactly the same, and below that a text thing with also the same size. This basically makes it look like you are working on an Excel sheet (or German bureaucracy form) where people don’t immediately know what the important stuff is and you get lost in a sea of things looking as if they have the same importance.


> One more thing though, aren't we also introducing this anti-pattern of having only an inline label: 

That’s another reason why the subject is differently styled: It’s very clear that it is the mail subject, without really needing a label. Also because (as said) it appears exactly the same as when you read Mail, so you have that symmetry. :)

skjnldsv commented:
> That’s another reason why the subject is differently styled: It’s very clear that it is the mail subject, without really needing a label. Also because (as said) it appears exactly the same as when you read Mail, so you have that symmetry. :)

Could we remove the bold though, it seems a bit too much.
Maybe juste the increased font size?

jancborchardt commented:
As said, the bold has been like that in Mail reading for some time now, and it looks really good giving point of reference. Also, we toyed around with giving the heading in the right sidebar bold, which is a h2 as well – you remember? ;)

So let’s keep it bold, and some other parts should be bold too.

ChristophWurst commented:
> **Travis CI - Pull Request ** Build Errored

Rule of thumb: make CI green *before* merging. This caused a build break on all new PRs and thus a huge Dependabot backlog.

jancborchardt commented:
@ChristophWurst sorry! Merged cause you gave me the green light via message. Forgot to run the linter, but am also confused cause I did not touch the content of the line – I guess it is because of the container and the further indent, and the linter wants it to not exceed a certain amount of characters?

ChristophWurst commented:
Could be, yeah
